sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosOperations.java

```diff
@@ -82,6 +83,27 @@ Mono<CosmosContainerProperties> replaceContainerProperties(String containerName,
      */
     <T> Flux<T> findAll(PartitionKey partitionKey, Class<T> domainType);
 
+    /**
+     * Find all items in a given container with partition key
+     *
+     * @param domainType    the domainType
+     * @param containerName the container name
+     * @param <T>           type of domainType
+     * @return results as Page
+     */
+    <T> CosmosPagedFlux<T> findAll(Class<T> domainType, String containerName);
+
+    /**
+     * Pagination query
+     *
+     * @param query         the document query
+     * @param domainType    type class
+     * @param containerName the container name
+     * @param <T>           type class of domainType
+     * @return results as Page
+     */
+    <T> CosmosPagedFlux<T> paginationQuery(CosmosQuery query, Class<T> domainType, String containerName);
```

Comment:
Do we need to expose containerName?
Also, this seems to be a very generic API that we are exposing, is there a customer request for this?

Reply:
For both these APIs, I wouldn't want to expose them directly in the repository if they are not required or requested. Since they are directly touching the public surface area. 
Also, what I understand from your feature is that now customers can create JPA queries with CosmosPagedFlux as return type. If that's correct, then these APIs can be added by customers themselves. 
We should mostly have only those APIs here which are generic enough and special at the same time to provide benefit to customer by some optimization, or fulfilling some necessity.